packages/webapp/src/components/Form/FormNavigationButtons/index.tsx

```diff
@@ -46,6 +50,18 @@ const FormNavigationButtons = ({
         <Button color="secondary-cta" onClick={onCancel} className={styles.button} sm fullLength>
           {t('common:CANCEL')}
         </Button>
+        {hasPreviousButton && (
```

Comment:
I think you can check if `onPrevious` exists like `contextualContent` is checked (line 48).

Reply:
Hi @SayakaOno , thank you for your suggestions :)

Using `onPrevious` to conditionally render the **Previous** button is a bit trickier than what it seems, since `An expression of type 'void' cannot be tested for truthiness` (a warning thrown by TypeScript). So two options come to my mind:

1. We make the type of `onPrevious` be `() => true` instead of `() => void`, forcing to `return true` any time we want to use the onPrevious prop;
2. We leave the `hasPreviousButton` prop.

Both options are a bit cumbersome and I don't like them much, but I didn't find a better solution. What do you think?

Reply:
Did you call `onPrevious()` to use it as the condition?
Without the `()`, TypeScript shouldn't complain!

Reply:
Storybook tricked me on this one. It is as you say: it works as it should without the `()`. What confused me was that every story now has the **Previous** button even though there is only one story (**WithPreviousButton**) that uses the `onPrevious` prop, so I thought that a void function was not valid for this use case. Nevertheless I tested the component on the app itself and the component works as it should.

I used Storybook in the past even though I'm not very familiarized with it, so if you have any suggestion to make the **Previous** button appear only on the **WithPreviousButton** story, please let me know.

Reply:
I am surprised we haven't run into that before! 

https://github.com/storybookjs/storybook/issues/22930

I think you could consider adding `onPrevious: undefined` to the variants . The overwriting args seems to be working correctly at least.

Reply:
It seems Storybook works fine only if it feels like it. I was working on a different ticket for which I also needed to run Storybook, then I changed to the local branch I'm using for this issue and stories were running as they should:

https://github.com/user-attachments/assets/9e56b37f-cc2f-45b3-9b6f-52a48fee2708

Then just out of curiosity I restarted Storybook and it started to show the Previous button on every story again 🤷 :

https://github.com/user-attachments/assets/07188110-2ae5-4853-89e3-132dbb292f98

I just added `onPrevious: undefined` to the variants as you proposed and now stories work fine too 👍